docs/arch/adr-03.md

```diff
@@ -0,0 +1,112 @@
+- Greater potential throughput
+
+#### Cons:
+- Launched in Sept 2015, hasn't been around for as long as Blob Storage.
+- Smaller max size (100 TiB: 1 TiB = 2^40 bytes). We are not expecting to come close to this size, so this isn’t much of a concern
+
+## Decision
+
+We don't appear to have any need for most of the additional functionality that comes with File Service, which makes me reluctant to want to use it. In addition, the number of articles and resources available on communicating with Blob Storage to set up file zipping is much greater than what's available for File Service. My initial understanding of Blob Storage led me to believe that permissions could only be set at the container level, but this turned out to be wrong. With the ability to set blob-specific permissions, we will be able to use a single container to store the transcription-specific files, and the user-requested zip files.
+
+Ultimately, my choice is to go with Blob Storage: the more basic, simple storage tool that gives us what we need and nothing more. That being said, I'd still like to keep the option of using Azure File Service on the table, in case it turns out that we *would* benefit from the additional functionality that it offers.
+
+As for what type of blob we will use, my choice would be to store each data file in its own block blob. If we were to choose to store multiple files within a single blob (and have each file be associated with a block ID on that blob), we would lose the ability to name each individual file. Hypothetically, it would be possible to create a database table with columns “block ID” and “name”, to emulate a naming functionality, but this seems far more complicated than its worth. In addition, the [azure-storage-blob](https://github.com/azure/azure-storage-ruby/tree/master/blob) gem gives us a simple interface for working with block blobs and saves us the trouble of having to write HTTP requests ourselves.
+
+Final questions:
+1. Q: Blob Storage doesn't have any concrete hierarchy beyond Storage Account/Blob Container - within a container, directories are virtual, demarcated by prefixes in the file name. Will this end up being problematic for us? Will it complicate file retrieval?  
+A: Retrieving files from a file system with virtual directories shouldn't be any different than retrieving files from a normal file system. As long as blob prefixes are constructed in a way that reflects the organizational system used within the application/database, there should be no trouble. File retrieval may be helped by append blobs - final decision on blob type is still TBD.
```

Comment:
Agree: I think this will be fine, since we have 100% predictable file paths and names. We shouldn't ever need a directory listing--which is good, since that doesn't seem possible. Good to keep in mind for using this service in the future, though.

Reply:
When you say directory listing, you mean a list of all folders within the blob container, right? I know that we're able to list all files in the container using `client.list_blobs(container.name)` (see [azure-storage-ruby](https://github.com/azure/azure-storage-ruby/tree/master/blob)) and then we'd be able to deduce what folders exist from that, but you're right it seems there is no way of requesting _just_ a list of folders directly.

Reply:
When you say directory listing, you mean a list of all folders within the blob container, right? I know that we're able to list all files in the container using `client.list_blobs(container.name)` (see [azure-storage-ruby](https://github.com/azure/azure-storage-ruby/tree/master/blob)) and then we'd be able to deduce what folders exist from that, but you're right it seems there is no way of requesting _just_ a list of folders directly.